### ISSUE TITLE
Torque sensor calibration instructions

### ISSUE THREAD

**Issue by bogardon** [emmebrusa/TSDZ2-Smart-EBike-860C#20]
Hi,

I've read through https://github.com/emmebrusa/TSDZ2-Smart-EBike-860C/wiki and https://github.com/OpenSourceEBike/TSDZ2_wiki/wiki/Features-and-configurations-on-display but I'm still not clear on the calibration instructions.

It seems that the original firmware is a lot more involved and requires taking 8 measurements per pedal for a total of 16. As far as I can tell, I should not follow those instructions for this firmware, is that right?

The instructions for this firmware don't cover everything I see under the Torque sensor section (I'm using v20.1C 5.0), for example:

### Undocumented fields

- Torque adc step adv
- Torque offset adj
- Torque range adj
- Torque angle adj
- ADC torque step calc

Can anybody clarify what those are? And if I need to touch them for calibration?

### Instructions

1. Go to `Technical` section and find `ADC torque sensor`.
2. Turn crank side pedal to 3 oclock position and record value of `ADC torque sensor` without any weight on it.
3. Hold the brakes and put all your weight on the crank side pedal in the same position and record new value of `ADC torque sensor`.
4. Go to `Torque sensor` section and enter value obtained in step 2 into `Torque adc offset`.
5. Enter value obtained in step 3 into `Torque adc max`.

Bonus steps:
1. In `Torque sensor` section, set `Weight on pedal` to `25`.
2. Hang 25kg of weight on the crank side pedal in 3 oclock position with brakes applied.
3. Go to `Technical` section and record value of  `ADC torque sensor`.
4. Go back to `Torque sensor` section and enter value obtained in step 3 into `Torque adc on weight`.
5. Set `Default weight` to `no`.

Are the above steps correct? Do I need to do anything with the non-crank side pedal? How do I correctly set the value of `Torque adc step`? Thanks in advance!

**Comment by bogardon** [comment]
I set `Torque adc offset` and `Torque adc max` according to my steps above and came out with 145 and 292, so I think I'm good in that there is >140 difference between them. 

However, I feel the human power calculations feel off. Reading further, I realized I had to complete the bonus steps with the 25kg weight, but before I try that (also still need clarification on how to do it), I tried to set `Default weight` to `yes`, but it keeps resetting back to `no` for some reason. Anyone know why?

**Comment by bogardon** [comment]
So I ended up doing the 25kg calibration. I actually used 24 for `weight on pedal` because for some reason whenever I confirm any value near 25, it just changes it to 24 anyways.

The human power is still very inaccurate. It underestimates by a lot, so much so that I thought it was more accurate without calibration.

See my attached torque sensor screen.

![IMG_0548](https://github.com/user-attachments/assets/d8e3f516-f035-496c-9f71-d092d8d8afb0)
